RHEL6_7/packages/RemovedPackages/noauto_postupgrade.d/native_rpm_dep_fix.sh

```diff
@@ -0,0 +1,135 @@
+    #
+    # Accepts one parameter, which says whether will be removed only native
+    # old RPMs with broken dependencies (default) or should be removed even
+    # non-native old RPMs with broken dependencies (when param=1)
+    #
+    # Returns 0 when no RPM has been removed. Otherwise returns 1.
+    #
+    local rm_custom_enabled=0
+    [ "$1" -eq 1 ] && rm_custom_enabled=1
+
+    get_broken_deps_list | grep "\.el6" > rpms_broken
+    local REMOVED_FLAG=0
+    local line=""
+    while IFS= read -r line || [ -n "$line" ]; do
+        NAME=$(rpm -q --qf '%{NAME}' $line)
+        is_dist_native "$NAME" || [ $rm_custom_enabled -eq 1 ] || {
+            # it's not dist native package -> skip it
+            echo "    $NAME" >> "$kept_broken_rpms"
+            log_info_verbose "Skip the 3rd party of custom $line RPM."
+            continue
+        }
+
+        # remove broken rpm
+        log_info_verbose "Removing the $line RPM."
+        rpm -e --nodeps "$line" || {
+            rpm -q "$line" >/dev/null 2>&1 || {
+                log_error "The $line RPM has not been removed."
+                FAIL=1
+            }
+            continue
+        }
+        REMOVED_FLAG=1
+    done < rpms_broken
+    return $REMOVED_FLAG
+}
+
+
+##################### MAIN ###########################
+
+# process params from cmdln
+while [[ -n "$1" ]]; do
+    case $1 in
+        -a | --rm-all)
+            RM_CUSTOM_RPM=1
+            ;;
+        -v | --verbose)
+            VERBOSE=1
+            ;;
+        -h | --help)
+            echo "USAGE: native_rpm_dep_fix [-a|--rm-all] [-v|--verbose] [-h|--help]"
+            echo "    -a | --rm-all   Enable removal of 3rd party or custom RPMs"
+            echo "    -v | --verbose  Activate verbose mode"
+            echo "    -h | --help     Print this help"
+            echo
+            exit 0
+            ;;
+    esac
+    shift
+done
+
+
+cd $(dirname "$0")
+rm -f "$kept_broken_rpms"
+touch "$kept_broken_rpms"
+
+[ $RM_CUSTOM_RPM -eq 1 ] \
+    && log_info_verbose "Removal of 3rd party or custom RPMs with broken dependencies has been enabled."
+
+for counter in {0..10}; do
+    # try just limited number of loops in most to be sure that scripts ends
+    # always (in reasonable time)
+    # .. in case that nothing is removed (returned 0) break the loop
+    rm_broken_old_rpms "$RM_CUSTOM_RPM" && break
+    [ $counter -eq 10 ] && {
+        [ -z "$(get_broken_deps_list)" ] && {
+            log_warning "Reached iteration limit but some RPMs are still broken."
+            log_info "You can run the application again to try to remove rest of RPMs."
+            FAIL=1
+        }
+    }
+done
+
+broken_nonnative_rpms=$(wc -l <"$kept_broken_rpms")
+[ $broken_nonnative_rpms -gt 0 ] && {
+    # this is obviously irrelevant in case of use the -a option
+    log_warning "Detected several 3rd party or custom RHEL 6 RPMs with broken dependencies:"
+    cat "$kept_broken_rpms" | sort | uniq >&2
```

Comment:
Useless use of cat :)

    <"$stuff" sort | uniq >&2

Reply:
I know. But this is only cat I really like, so I put it on many places :heart: